rust/lance/src/dataset/transaction.rs

```diff
@@ -250,10 +259,12 @@ impl Transaction {
                 &other.operation,
                 Operation::Overwrite { .. } | Operation::Restore { .. }
             ),
-            Operation::CreateIndex { .. } => match &other.operation {
+            Operation::CreateIndex { .. } | Operation::ModifyIndex { .. } => match &other.operation
+            {
                 Operation::Append { .. } => false,
                 // Indices are identified by UUIDs, so they shouldn't conflict.
                 Operation::CreateIndex { .. } => false,
+                Operation::ModifyIndex { .. } => false,
```

Comment:
if two ops are mofifying the same index they should conflict right?

Reply:
txn log is ordered , right? it means modifying index then recreate an index over it?

Reply:
I think I agree with @eddyxu .  We do the same thing, for example, with overwrite / restore.  Both operations succeed even though one of them is immediately obsoleted.

Reply:
I think modify index should conflict with modify index, if they modify the same index. I think it is compatible with CreateIndex (unless CreateIndex is overwriting the index it modifies).

These are transactions that happened at the same time, so you can't commit modified files if the files those are based on have been modified concurrently.